plugin/storage/es/factory.go

```diff
@@ -190,32 +192,56 @@ func createSpanWriter(
 }
 
 // GetSpanServiceMappings returns span and service mappings
-func GetSpanServiceMappings(shards, replicas int64, esVersion uint) (string, string) {
+func GetSpanServiceMappings(shards, replicas int64, esVersion uint, esPrefix string, useILM bool) (string, string, error) {
 	if esVersion == 7 {
-		return fixMapping(loadMapping("/jaeger-span-7.json"), shards, replicas),
-			fixMapping(loadMapping("/jaeger-service-7.json"), shards, replicas)
+		spanMapping, er := fixMapping(es.PongoTemplateBuilder{}, loadMapping("/jaeger-span-7.json"), shards, replicas, esPrefix, useILM)
+		if er != nil {
+			return "", "", er
```

Comment:
Added tests for these, but unfortunately could not use Once() - somehow it was not working out for me, even with Once() specified - same mock execute was being called. Might need help on this.

Reply:
The reason why `Once()` not working is because you're creating a new mock each time [`newTextTemplateBuilder()`](https://github.com/jaegertracing/jaeger/pull/2454/files#diff-bd7140c4b097bbb7e2462f2b512528d3f45af2317fe008211c293553bff71c56R199) is called, which means the second mocked return is never exercised.

See: https://github.com/jaegertracing/jaeger/pull/2454#discussion_r559404668 for how to address the problem.